core/services/relay/evm/codec.go

```diff
@@ -0,0 +1,124 @@
+package evm
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"reflect"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/codec"
+
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+// decodeAccountHook allows strings to be converted to [32]byte allowing config to represent them as 0x...
+// EpochToTimeHook allows times to be converted to and from integer types
+// BigIntHook allows *big.Int to be represented as any integer type or a string and to go back to them.
+// Useful for config, or if when a model may use a go type that isn't a *big.Int when Pack expects one.
+// Eg: int32 in a go struct from a plugin could require a *big.Int in Pack for int24, if it fits, we shouldn't care.
+// SliceToArrayVerifySizeHook verifies that slices have the correct size when converting to an array
+// sizeVerifyBigIntHook allows our custom types that verify the number fits in the on-chain type to be converted as-if
+// it was a *big.Int
+var evmDecoderHooks = []mapstructure.DecodeHookFunc{decodeAccountHook, codec.EpochToTimeHook, codec.BigIntHook, codec.SliceToArrayVerifySizeHook, sizeVerifyBigIntHook}
```

Comment:
Is all this decoding/encoding for passing args through LOOPS?

Reply:
Not LOOPPs exactly. Just the relayer interface. It has only been chain-agnostic, but this moves it towards being product agnostic.